ovirt_imageio/_internal/server.py

```diff
@@ -114,15 +117,16 @@ def configure_logger(cfg):
 
 class Server:
 
-    def __init__(self, config):
+    def __init__(self, config, ticket=None):
         self.config = config
         self.running = False
         self.auth = auth.Authorizer(config)
         self.remote_service = services.RemoteService(self.config, self.auth)
         self.local_service = None
         if config.local.enable:
             self.local_service = services.LocalService(self.config, self.auth)
-        self.control_service = services.ControlService(self.config, self.auth)
+        self.control_service = services.ControlService(
+            self.config, self.auth, ticket)
```

Comment:
No need to modify the control service - we initialize the service with the auth instance. The only thing we need to do at this point is to add the ticket to the auth:

    self.auth.add(read_ticket(ticket))

And no change is needed in the control service - which can and should be disable when using containerize deployment!